sql/catalyst/src/main/java/org/apache/spark/sql/catalog/v2/CaseInsensitiveStringMap.java

```diff
@@ -0,0 +1,107 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.spark.sql.catalog.v2;
```

Comment:
shall we move it to `org.apache.spark.sql.v2`? so that the non-catalog part can also use it.

Reply:
ah there is no `org.apache.spark.sql.v2`, we may need to revist the package name and see if there can be a shared parent package for sources and catalogs.

Reply:
For now maybe just put this class in `org.apache.spark.sql`?

Reply:
If we don't have `v2`, can we put into `org.apache.spark.sql.util` like Scala `CaseInsensitiveMap`?

Reply:
I like `org.apache.spark.sql.util`. I'll move it there.